monai/transforms/inverse.py

```diff
@@ -227,17 +223,16 @@ def track_transform_meta(
             if out_obj.pending_operations:
                 transform_name = info.get(TraceKeys.CLASS_NAME, "") if isinstance(info, dict) else ""
                 msg = (
-                    f"Applying transform {transform_name} to a MetaTensor with pending operations "
-                    "is not supported (as this eventually changes the ordering of applied_operations when the pending "
-                    f"operations are executed). Please clear the pending operations before transform {transform_name}."
-                    f"\nPending operations: {[x.get(TraceKeys.CLASS_NAME) for x in out_obj.pending_operations]}."
+                    f"Transform {transform_name} has been applied to a MetaTensor with pending operations: "
+                    f"{[x.get(TraceKeys.CLASS_NAME) for x in out_obj.pending_operations]}"
                 )
+                if key is not None:
+                    msg += f" for key {key}"
+
                 pend = out_obj.pending_operations[-1]
-                if not isinstance(pend.get(TraceKeys.EXTRA_INFO), dict):
-                    pend[TraceKeys.EXTRA_INFO] = dict(pend.get(TraceKeys.EXTRA_INFO, {}))
-                if not isinstance(info.get(TraceKeys.EXTRA_INFO), dict):
-                    info[TraceKeys.EXTRA_INFO] = dict(info.get(TraceKeys.EXTRA_INFO, {}))
-                info[TraceKeys.EXTRA_INFO]["warn"] = pend[TraceKeys.EXTRA_INFO]["warn"] = msg
+                reasons = pend.get(TraceKeys.INVERT_DISABLED, list())
```

Comment:
I don't think the flag name `INVERT_DISABLED` is appropriate here. at this point this is not just an 'invert' related issue, maybe some user-specified transforms accidentally run into this logic, and the result is also not just about 'inverse', it potentially affects any operations that rely on the ordering of applied_operations.

Reply:
What are the other circumstances?

Reply:
these `TraceableTransform` and `applied_operations` are generic APIs in monai, the user may call `track_transform_meta` in their own transform, may read `applied_operations` in their applications and process their business logic.

Reply:
That doesn't matter. For whatever reason that has happened, if someone pushes an `applied_operation` while there are `pending_operations`, the order has changed and the pipeline can't be inverted by `Compose.inverse`.

Reply:
it matters because invertible is just one of the possible applications of traceable

Reply:
In that case, the logic shouldn't be part of that function, but should be handled outside of the function.

Reply:
@wyli There is now a STATUS TraceKey and a new enum TraceStatusKeys that currently contains PENDING_DURING_APPLY.